File: 1122-relative-sort-array/1122-relative-sort-array.cpp
class Solution {
public:
    vector<int> relativeSortArray(vector<int>& arr1, vector<int>& arr2) {
        map<int, int>mp;
        for(int x:arr1) mp[x]++;
        vector<int>ans;
        for(int x:arr2) {
            while(mp[x] > 0) {
                ans.push_back(x);
                mp[x]--;
            }
        }
        for(auto it:mp) {
            while(it.second > 0) {
                ans.push_back(it.first);
                it.second--;
            }
        }
        return ans;
    }
};

File: 1122-relative-sort-array/1122-relative-sort-array.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        Map<Integer, Integer>mp = new TreeMap<>();
        for(int x:arr1) mp.put(x, mp.getOrDefault(x, 0) + 1);
        int ans[] = new int[arr1.length];
        int ind = 0;
        for(int x:arr2) {
            if(mp.containsKey(x)) {
                while(mp.get(x) > 0) {
                    ans[ind++] = x;
                    mp.put(x, mp.get(x) - 1);
                }
            }
        }
        for(int x:mp.keySet()) {
            while(mp.get(x) > 0) {
                ans[ind++] = x;
                mp.put(x, mp.get(x) - 1);
            }
        }
        return ans;
    }
}